charts/vault-monitoring/templates/endpoint.yaml

```diff
@@ -0,0 +1,15 @@
+{{- if eq .Values.vault.service.type "externalName" }}
```

Comment:
Maybe this should have an `else` that fired an error when type is non empty and not "externalName"?

Reply:
* [ ] TODO

Reply:
There is no endpoint needed if the service is not externalName. The endpoint is created by the label matching.

Reply:
If you try specifying an invalid `externalName` an error would be nice. It should still allow deactivating the endpoint if you don't specify one.

Reply:
I don't understand. Please suggest the edit so we can move on. Thanks.

Reply:
I was thinking something like this instead of just `{{ end }}`.
```
...
{{- else if not (empty .Values.vault.service.type) }}
{{ fail (printf "Invalid service type: %s" .Values.vault.service.type) }}
{{- end }}
```

Reply:
I was thinking something like this instead of just `{{ end }}`.
```
...
{{- else if not (empty .Values.vault.service.type) }}
{{ fail (printf "Invalid service type: %s" .Values.vault.service.type) }}
{{- end }}
```

Reply:
I don't get it. What if the `vault.service.type` is `clusterIP`? why should it fail? 

`externalName` => create Endpoint + Service 
`clusterIP` => create just Service (with selectors).

Reply:
I see, I'm looking at [the docs](https://kubernetes.io/docs/concepts/services-networking/service/#externalname) and this doesn't feel as consistent as I though it should be. I guess it's fine but I don't really understand the following:
* Why isn't it `vault.service.type=ExternalName` instead of `vault.service.type=externalName`?
* Can't we just specify a canonical DNS name directly in the service and not manage this endpoint manually?

Reply:
>     * Why isn't it `vault.service.type=ExternalName` instead of `vault.service.type=externalName`?

```yaml
type: ExternalName
```
is correct: https://kubernetes.io/docs/concepts/services-networking/service/#externalname

>     * Can't we just specify a canonical DNS name directly in the service and not manage this endpoint manually?

Prometheus ServiceMonitor needs an endpoint to work.  A lone Service without endpoint would not work for ServiceMonitor.